static/app/utils/useRequest.tsx

```diff
@@ -0,0 +1,252 @@
+   * result set using a simple search query.
+   *
+   * Will always add new options into the store.
+   */
+  onSearch: (searchTerm: string) => Promise<void>;
+  nextPage: () => Promise<void>;
+  get: () => Promise<void>;
+  post: (_) => Promise<void>;
+  update: (_) => Promise<void>;
+  del: () => Promise<void>;
+} & State;
+
+type Options = {
+  /**
+   * Number of items to return per page
+   */
+  limit?: number;
+};
+
+type FetchRequestOptions = {
+  limit?: Options['limit'];
+  cursor?: State['nextCursor'];
+  search?: State['lastSearch'];
+  lastSearch?: State['lastSearch'];
+};
+
+/**
+ * Helper function to actually load items
+ */
+async function fetch(
+  api: Client,
+  url: string,
+  {search, limit, lastSearch, cursor}: FetchRequestOptions = {}
+) {
+  const query: {
+    query?: string;
+    cursor?: typeof cursor;
+    per_page?: number;
+  } = {};
+
+  if (search) {
+    query.query = `${query.query ?? ''} ${search}`.trim();
+  }
+
+  const isSameSearch = lastSearch === search || (!lastSearch && !search);
+
+  if (isSameSearch && cursor) {
+    query.cursor = cursor;
+  }
+
+  if (limit !== undefined) {
+    query.per_page = limit;
+  }
+
+  let hasMore: null | boolean = false;
+  let nextCursor: null | string = null;
+  const [data, , resp] = await api.requestPromise(url, {
+    includeAllArgs: true,
+    query,
+  });
+
+  const pageLinks = resp?.getResponseHeader('Link');
+  if (pageLinks) {
+    const paginationObject = parseLinkHeader(pageLinks);
+    hasMore = paginationObject?.next?.results || paginationObject?.previous?.results;
+    nextCursor = paginationObject?.next?.cursor;
+  }
+
+  return {results: data, hasMore, nextCursor};
+}
+
+function useRequest(
+  url: string,
+  {limit}: Options = {},
+  onSuccessCb = _ => {},
+  onErrorCb = _ => {}
+) {
+  const api = useApi();
+  const {organization} = useLegacyStore(OrganizationStore);
+
+  const orgId = organization?.slug;
+
+  const [state, setState] = useState<State>({
+    data: null,
+    loading: false,
+    hasMore: null,
+    lastSearch: null,
+    nextCursor: null,
+    error: null,
+  });
+
+  async function get(cursor: string | null = null) {
+    setState({...state, loading: true});
+    try {
+      const {results, hasMore, nextCursor} = await fetch(api, url, {
+        limit,
+        cursor,
+      });
+      onSuccessCb(results);
+      setState({
+        ...state,
+        data: results,
+        hasMore,
+        loading: false,
+        nextCursor,
+      });
+    } catch (err) {
+      Sentry.captureException(err);
+      setState({...state, loading: false, error: err});
+      onErrorCb(err);
+    }
+  }
+
+  async function handleSearch(search: string) {
+    const {lastSearch} = state;
+    const cursor = state.nextCursor;
+
+    if (search === '') {
+      return;
+    }
+
+    if (orgId === undefined) {
+      // eslint-disable-next-line no-console
+      console.error('Cannot use useRequest.onSearch without an organization in context');
```

Comment:
Why?

Reply:
yea you're right, we don't need organization here. This was a legacy from the useRepositories hook pre-splitting up the PR.